### PR TITLE
General Settings tweaks.

### PR DESCRIPTION
**Pull request by dogmaphobic**

### CODE REVIEW

**Comment by DonLakeFlyer** [comment]
This whole thing needs to be reworked to not use fixed widths. It currently blows up on small screens. So adding more won't help.

File: src/ui/preferences/GeneralSettings.qml [hunk 1]
```diff
@@ -140,7 +140,7 @@ QGCView {
                         //-----------------------------------------------------------------
                         //-- Base UI Font Point Size
                         Row {
-                            visible: _appFontPointSize.visible
+                            visible: _appFontPointSize ? _appFontPointSize.visible : false
```
Comment (DonLakeFlyer) [inline src/ui/preferences/GeneralSettings.qml]: Why is this needed? This should never be missing.

Comment (dogmaphobic) [inline src/ui/preferences/GeneralSettings.qml]: It was throwing a Javascript exception because _appFontPointSize wasn't defined.

**Comment by DonLakeFlyer** [comment]
It's also going to conflict like crazy with the stuff I'm doing with save paths already.

File: src/ui/preferences/GeneralSettings.qml (hunk 1 above)
Comment (DonLakeFlyer) [inline src/ui/preferences/GeneralSettings.qml]: That's a bug somewhere then. Settings facts should never be missing.
